### PR TITLE
Add checksum for py-seaborn 0.12.0

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-seaborn/package.py [hunk 1]
```diff
@@ -17,26 +17,32 @@ class PySeaborn(PythonPackage):
...
+    depends_on("py-flit-core@3.2:3", when="@0.12.0:", type="build")
 
+    depends_on("py-numpy@1.17:", when="@0.12:", type=("build", "run"))
     depends_on("py-numpy@1.15:", when="@0.11:", type=("build", "run"))
     depends_on("py-numpy@1.9.3:", when="@0.9:", type=("build", "run"))
     depends_on("py-numpy", type=("build", "run"))
     depends_on("py-scipy@1:", when="@0.11:", type=("build", "run"))
     depends_on("py-scipy@1.0.1:", when="@0.10:", type=("build", "run"))
     depends_on("py-scipy@0.14:", when="@0.9.0:", type=("build", "run"))
     depends_on("py-scipy", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-seaborn/package.py]: Looks like scipy is an optional dep now. I would add upper bounds on these, and optionally add a new stats variant with deps on scipy and statsmodels.

File: var/spack/repos/builtin/packages/py-seaborn/package.py [hunk 2]
```diff
@@ -17,26 +17,32 @@ class PySeaborn(PythonPackage):
     homepage = "https://seaborn.pydata.org/"
     pypi = "seaborn/seaborn-0.7.1.tar.gz"
 
+    version("0.12.0", sha256="893f17292d8baca616c1578ddb58eb25c72d622f54fc5ee329c8207dc9b57b23")
     version("0.11.2", sha256="cf45e9286d40826864be0e3c066f98536982baf701a7caa386511792d61ff4f6")
     version("0.11.1", sha256="44e78eaed937c5a87fc7a892c329a7cc091060b67ebd1d0d306b446a74ba01ad")
     version("0.9.0", sha256="76c83f794ca320fb6b23a7c6192d5e185a5fcf4758966a0c0a54baee46d41e2f")
     version("0.7.1", sha256="fa274344b1ee72f723bab751c40a5c671801d47a29ee9b5e69fcf63a18ce5c5d")
 
     depends_on("python@3.6:", when="@0.10:", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-seaborn/package.py:28]: Python 3.7+ is required now

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-seaborn/package.py [hunk 3]
```diff
@@ -17,26 +17,41 @@ class PySeaborn(PythonPackage):
...
     depends_on("py-matplotlib@2.2:", when="@0.11:", type=("build", "run"))
     depends_on("py-matplotlib@2.1.2:", when="@0.10:", type=("build", "run"))
     depends_on("py-matplotlib@1.4.3:", when="@0.9:", type=("build", "run"))
     depends_on("py-matplotlib", type=("build", "run"))
+
+    depends_on("py-scipy@1:", when="@0.12: +stats", type=("build", "run"))
+    depends_on("py-scipy@1:", when="@0.11:", type=("build", "run"))
+    depends_on("py-scipy@1.0.1:", when="@0.10:", type=("build", "run"))
+    depends_on("py-scipy@0.14:", when="@0.9.0:", type=("build", "run"))
+    depends_on("py-scipy", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-seaborn/package.py]: These need upper bounds, otherwise scipy will be installed when `@0.12 ~stats`

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Comment by iarspider** [comment]
@spackbot fix style

**Review by adamjstewart — DISMISSED** [review]

**Review by adamjstewart — DISMISSED** [review]

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**